vcr-workflow.Rmd

```diff
@@ -0,0 +1,66 @@
+# vcr workflows {#vcr-workflows}
+
+- Re-recording cassettes
+- Making sure vcr is working
+- Make tasks
+
+## Re-recording cassettes {#vcr-re-recording}
+
+There are a number of reasons why you may want to re-record your cassettes:
+
+1. `vcr`/`webmockr` upgrades: you may want to re-record your cassettes if
+you've recently upgraded either of these packages. There should in general
+be no problem if you do not re-record your cassettes when using new versions
+of `vcr`/`webmockr`. Cassettes record the version of `vcr`/`webmockr` they 
+were recorded with.
+2. Code linked to cassette has changed in a way that would lead to different
+request/response. Whenever your code has changed leading to different request
+and/or response you should record a new cassette. 
+3. Remote API has changed leading to a different response. In this case the
+HTTP request is likely identical to the one on the cassette, but since the 
+remote API response has changed, you should record a new cassette. Note that
+matching HTTP requests is not done on HTTP responses, but if your tests 
+check any aspect of the response then you'll likely want to re-record 
+your cassettes.
+
+How do you re-record a cassette?
+
+This can be as simple as deleting existing cassettes and running tests again,
+which should record new cassettes.
+
+Another option is to use the vcr configuration option [re_record_interval][rri].
+With this option you can set a time in seconds after which a new cassette
+will be recorded, deleting the old one first. This will help make sure your
+cassettes have up to date HTTP response bodies from the remote API; helping
+you avoid cases where you're essentially testing old API behavior.
+
+## Making sure vcr is working {#vcr-working}
+
+How do you know vcr is working without internet access?
+
+- Make sure you have recorded cassettes. Then turn WI-FI off. Now run tests again.
```

Comment:
Yes that tip is also earlier in the book because it's an excellent tip you had given me!

Reply:
oh good. okay to duplicate here?

Reply:
yes!